connector/src/main/java/com/google/cloud/hive/bigquery/connector/output/indirect/IndirectAvroRecordWriter.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2022 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.hive.bigquery.connector.output.indirect;
+
+import com.google.cloud.hive.bigquery.connector.BigQuerySerDe;
+import com.google.cloud.hive.bigquery.connector.JobInfo;
+import com.google.cloud.hive.bigquery.connector.utils.HiveUtils;
+import com.google.cloud.hive.bigquery.connector.utils.avro.AvroDeserializer;
+import com.google.cloud.hive.bigquery.connector.utils.avro.AvroUtils.AvroOutput;
+import java.io.IOException;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.ql.exec.FileSinkOperator;
+import org.apache.hadoop.hive.serde2.objectinspector.StructObjectInspector;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.io.ObjectWritable;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mapred.JobConf;
+import org.apache.hadoop.mapred.Reporter;
+import org.apache.hadoop.mapred.TaskAttemptID;
+
+/**
+ * Writes records to an Avro file in GCS. Each task runs its own instance of this writer class, i.e.
+ * each task creates a single Avro file. The overall job committer is responsible for loading all
+ * the Avro files to BigQuery later on at the end of the job.
+ */
+public class IndirectAvroRecordWriter
```

Comment:
Why aren't we using the built in `org.apache.hadoop.hive.ql.io.avro.AvroContainerOutputFormat` ?

Reply:
This is so we can control the format of Avro files so that they preserve all the original data when it gets loaded to BQ. I had found that some details were lost or corrupted otherwise when using the Hive's built-in Avro serializer. E.g. if I recall correctly, the nano seconds were lost from timestamps.